src/generate.rs

```diff
@@ -199,18 +200,197 @@ pub fn cmd(opts: &Opts) -> exitcode::ExitCode {
     }
 
     if !errs.is_empty() {
-        errs.iter().for_each(|e| eprintln!("Generate error: {}", e));
-        return exitcode::CONFIG;
+        return Err(errs);
+    }
+
+    let mut builder = match toml::to_string(&globals) {
+        Ok(s) => s,
+        Err(err) => {
+            errs.push(format!("failed to marshal globals: {}", err));
+            return Err(errs);
+        }
+    };
+    if let Some(sources) = config.sources {
+        match toml::to_string(&{
+            let mut sub = Config::default();
+            sub.sources = Some(sources);
+            sub
+        }) {
+            Ok(v) => builder = [builder, v].join("\n"),
+            Err(e) => errs.push(format!("failed to marshal sources: {}", e)),
+        }
+    }
+    if let Some(transforms) = config.transforms {
+        match toml::to_string(&{
+            let mut sub = Config::default();
+            sub.transforms = Some(transforms);
+            sub
+        }) {
+            Ok(v) => builder = [builder, v].join("\n"),
+            Err(e) => errs.push(format!("failed to marshal transforms: {}", e)),
+        }
+    }
+    if let Some(sinks) = config.sinks {
+        match toml::to_string(&{
+            let mut sub = Config::default();
+            sub.sinks = Some(sinks);
+            sub
+        }) {
+            Ok(v) => builder = [builder, v].join("\n"),
+            Err(e) => errs.push(format!("failed to marshal sinks: {}", e)),
+        }
+    }
+
+    if !errs.is_empty() {
+        Err(errs)
+    } else {
+        Ok(builder)
     }
+}
 
-    match toml::to_string_pretty(&config) {
+pub fn cmd(opts: &Opts) -> exitcode::ExitCode {
+    match generate_example(&opts.expression) {
         Ok(s) => {
             println!("{}", s);
             exitcode::OK
         }
-        Err(e) => {
-            eprintln!("Failed to generate config: {}.", e);
+        Err(errs) => {
+            errs.iter().for_each(|e| eprintln!("Generate error: {}", e));
             exitcode::SOFTWARE
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // !IF YOU MODIFIED COMPONENT FIELDS AND NOW THESE TESTS FAIL!
```

Comment:
I don't really think this is necessary. `generate` is a maturing feature and our team should start to care about it the same way they do other parts of the code. Let folks get annoyed by this, so we get motivated to find a better way. :)

Reply:
True, fair enough.